video/enhanced_samples_generator.h

```diff
@@ -14,37 +16,46 @@
 class EnhancedSamplesGenerator : public WaveformGenerator {
 	public:
 		EnhancedSamplesGenerator(std::shared_ptr<AudioSample> sample);
+		~EnhancedSamplesGenerator();
 
 		void setFrequency(int value);
 		void setSampleRate(int value);
 		int getSample();
+		void setSample(std::shared_ptr<AudioSample> sample);
 
 		int getDuration(uint16_t frequency);
 
 		void seekTo(uint32_t position);
 	private:
-		std::weak_ptr<AudioSample> _sample;
+		// std::weak_ptr<AudioSample> _sample;
+		std::shared_ptr<AudioSample> _sample;
```

Comment:
it seems that using a weak_ptr here is problematic, and can occasionally (rarely) cause incorrect reference counting which can results in the AudioSample object being deleted, causing a crash

Reply:
simplifications made